src/ZODB/Connection.py

```diff
@@ -757,7 +758,11 @@ def afterCompletion(self, transaction):
         #    finalizing previous ones without calling begin.  We pass
         #    False to avoid possiblyt expensive sync calls to not
         #    penalize well-behaved applications that call begin.
-        self.newTransaction(transaction, False)
+        if hasattr(self._storage, 'afterCompletion'):
+            self._storage.afterCompletion()
```

Comment:
This is going to result in deferring `storage.sync`, `storage.poll_invalidations` and `cache.incrgc` for non-explicit transactions. It seems like the reasoning for point (a) should still apply though, and so maybe those things should still be done before `afterCompletion`?

[In the RelStorage case (post zopefoundation/relstorage#167) where explicit transactions *are* on, on the surface this looks like it would result in rolling back the load connection twice, but it actually won't because RelStorage handles that case already.]